src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final String SUPPLY_NAME = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String[] linePart;
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+            String value = reader.readLine();
+
+            while (value != null) {
+                linePart = value.split(",");
+                if (linePart[0].equals(SUPPLY_NAME)) {
+                    supplyAmount += Integer.parseInt(linePart[1]);
```

Parsing is done with `Integer.parseInt(linePart[1])` without handling parse errors. The checklist warns: "Don't ignore exceptions
Leaving an empty catch block or `e.printStackTrace` here is a bad practice. Better re-throw `RuntimeException` with the original exception and some message in the parameters:". If `amount` is not an integer you should catch NumberFormatException (or catch around parsing) and throw a `RuntimeException` with a clear message and the original exception as the cause.